### PR TITLE
feat: show version info

### LINKED ISSUE #13: Add option for showing version information
A CLI option `--version` or `-V` that displays the version information of commitlint.

### PR DESCRIPTION
**Pull request by aj3sh**
## Description

Shows version information on `-V` and `--version`.

## Related Issue

Fixes #13 

## Type of Change

Please mark the appropriate option below to describe the type of change your pull request introduces:

-   [ ] Bug fix
-   [x] New feature
-   [ ] Enhancement
-   [ ] Documentation update
-   [ ] Refactor
-   [ ] Other (please specify)

## Checklist

-   [x] My pull request has a clear title and description.
-   [x] I have used [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages).
        Examples: `"fix: Fixed foobar bug"`, `"feat(accounts): Added foobar feature"`.
-   [ ] I have added/updated the necessary documentation on `README.md`.
-   [ ] I have added appropriate test cases (if applicable) to ensure the changes are functioning correctly.

## Additional Notes

N/A

By submitting this pull request, I confirm that I have read and complied with the contribution guidelines of this project.

### CODE REVIEW

File: src/commitlint/__version__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+"""
+This module contains the version information of the current commitlint.
+TODO: automatically bump version through CI.
+"""
+
+__version__ = "0.2.1"
```
Comment (sugat009) [inline src/commitlint/__version__.py:6]: question: Is `setup.cfg` reusable for this use-case?

**Review by sugat009 — CHANGES_REQUESTED** [review]

File: src/commitlint/__version__.py (hunk 1 above)
Comment (aj3sh) [reply src/commitlint/__version__.py:6]: `setup.cfg` can be used for version information. However, the problem lies in the expense of reading the configuration, making it inefficient. 

Many standard libraries follow similar practices. Here are some references.

mypy: https://github.com/python/mypy/blob/c26f1297d4f19d2d1124a30efc97caebb8c28616/mypy/version.py#L11
pylint: https://github.com/pylint-dev/pylint/blob/2088414ef7353f98b65f61ff211150d9d7816c89/pylint/__pkginfo__.py#L12
celery: https://github.com/celery/celery/blob/7528900d53875bb634f92229ec6ed915c6fc58b6/celery/__init__.py#L20
flake8: https://github.com/PyCQA/flake8/blob/5c52d752e64f210369b37654e7b26ce25e2266d6/src/flake8/__init__.py#L20

Comment (sugat009) [reply src/commitlint/__version__.py:6]: My point is that if `__version__.py` file is created with version information then, there will be two places where version information exists, `__version__.py` file and `setup.cfg` file. In the case of the above-linked libraries, either they directly put their version in a setup file or reuse the `__version__.py` file in the setup file which we could do as well.

Comment (aj3sh) [reply src/commitlint/__version__.py:6]: I have found the similar problem in the stackoverflow: https://stackoverflow.com/questions/72357031/set-version-of-module-from-a-file-when-configuring-setuptools-using-setup. Let me check it.

**Review by sugat009 — APPROVED** [review]

File: src/commitlint/__version__.py (hunk 1 above)
Comment (sugat009) [reply src/commitlint/__version__.py:6]: Reviewed. Thanks @aj3sh . Over to you @subashcs .

**Review by subashcs — APPROVED** [review]
Looks good

File: src/commitlint/__version__.py (hunk 1 above)
Comment (subashcs) [reply src/commitlint/__version__.py:6]: Looks good

**Pull request merged**